server/controllers/controllers.js

```diff
@@ -18,12 +18,43 @@ exports.getTopics = (req, res, next) => {
 };
 
 exports.getArticles = (req, res, next) => {
-  fetchArticles()
-    .then((articles) => {
-      res.status(200).send({ articles });
+    if (req.query.hasOwnProperty("sort_by")) {
+        
+        const validSortBy = ["article_id", "title", "topic", "author", "body", "created_at", "votes", "article_img_url"];
+
+            if (!validSortBy.find(column => column === req.query.sort_by)) {
+                next("That column doesn't exist");
```

Comment:
The controller should be catching exceptions that have been thrown by the model so I would suggest that these checks are better performed there. If they fail, they can be handled with a Promise rejection that is caught in the controller. You can pass you error object to the rejection and handle it in your custom error handler.